code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,837 @@
+	id = "ephedrine"
+	result = "ephedrine"
+	required_reagents = list("sugar" = 1, "oil" = 1, "hydrogen" = 1, "diethylamine" = 1)
+	result_amount = 4
+	mix_message = "The solution fizzes and gives off toxic fumes."
+
+datum/reagent/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	description = "Purges body of lethal Histamine and reduces jitteriness while causing minor drowsiness."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+datum/reagent/diphenhydramine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.drowsyness += 1
+	M.jitteriness -= 1
+	M.reagents.remove_reagent("histamine",3)
+	..()
+	return
+
+/datum/chemical_reaction/diphenhydramine
+	name = "Diphenhydramine"
+	id = "diphenhydramine"
+	result = "diphenhydramine"
+	required_reagents = list("oil" = 1, "carbon" = 1, "bromine" = 1, "diethylamine" = 1, "ethanol" = 1)
+	result_amount = 4
+	mix_message = "The mixture dries into a pale blue powder."
+
+datum/reagent/morphine
+	name = "Morphine"
+	id = "morphine"
+	description = "Will allow you to ignore slowdown from equipment and damage. Will eventually knock you out if you take too much. If overdosed it will cause jitteriness, dizziness, force the victim to drop items in their hands and eventually deal toxin damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	var/cycle_count = 0
+	overdose_threshold = 30
+	addiction_threshold = 25
+
+
+datum/reagent/morphine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.status_flags |= IGNORESLOWDOWN
+	if(cycle_count >= 36)
+		M.sleeping += 3
+	cycle_count++
+	..()
+	return
+
+datum/reagent/morphine/overdose_process(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.Dizzy(1)
+		M.Jitter(1)
+	..()
+	return
+
+datum/reagent/morphine/addiction_act_stage1(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.Dizzy(2)
+		M.Jitter(2)
+	..()
+	return
+datum/reagent/morphine/addiction_act_stage2(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(1*REM)
+		M.Dizzy(3)
+		M.Jitter(3)
+	..()
+	return
+datum/reagent/morphine/addiction_act_stage3(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(2*REM)
+		M.Dizzy(4)
+		M.Jitter(4)
+	..()
+	return
+datum/reagent/morphine/addiction_act_stage4(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(3*REM)
+		M.Dizzy(5)
+		M.Jitter(5)
+	..()
+	return
+
+datum/reagent/oculine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	cycle_amount++
+	if(M.eye_blind > 0 && cycle_amount > 20)
+		if(prob(30))
+			M.eye_blind = 0
+		else if(prob(80))
+			M.eye_blind = 0
+			M.eye_blurry = 1
+		if(M.eye_blurry > 0)
+			if(prob(80))
+				M.eye_blurry = 0
+	..()
+	return
```

Comment:
Due to how we have internal organs and jazz, I recommend using this for oculine instead: https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/reagents/Chemistry-Reagents.dm#L1758